src/io/flutter/logging/FlutterLogTree.java

```diff
@@ -128,37 +129,48 @@ void render(FlutterLogEntry entry) {
     }
 
     private class MessageCellRenderer extends EntryCellRenderer {
-      @Nullable
-      private final FlutterConsoleFilter consoleFilter;
+      @NotNull
+      private final FlutterApp app;
+      @NotNull
+      private final List<Filter> linkFilters;
 
-      MessageCellRenderer(@Nullable Module module) {
-        consoleFilter = module != null ? new FlutterConsoleFilter(module) : null;
+      MessageCellRenderer(@NotNull FlutterApp app) {
+        this.app = app;
+        linkFilters = new ArrayList<>();
+        if (app.getModule() != null) {
+          linkFilters.add(new FlutterConsoleFilter(app.getModule()));
+        }
+        linkFilters.addAll(Arrays.asList(
+          new DartConsoleFilter(app.getProject(), app.getProject().getBaseDir()),
+          new UrlFilter()
+        ));
       }
 
       @Override
       void render(FlutterLogEntry entry) {
         // TODO(pq): SpeedSearchUtil.applySpeedSearchHighlighting
         // TODO(pq): setTooltipText
         final String message = entry.getMessage();
-        int cursor = 0;
+        final List<Filter.ResultItem> resultItems = linkFilters.stream()
+          .map(filter -> filter.applyFilter(message, message.length()))
+          .filter(Objects::nonNull)
+          .flatMap(result -> result.getResultItems().stream())
+          .sorted(Comparator.comparingInt(Filter.ResultItem::getHighlightStartOffset))
+          .collect(Collectors.toList());
```

Comment:
Oh cool!

One concern: might we consider performance here?  There's a bit of conflicting information about the impact of streams on performance and though I'm generally skeptical of micro-benchmarks, this is a case where I think we should proceed cautiously.  As we all know, `render(..)` gets called very frequently and we should be sure it's very performant with huge lists of entries.

My gut says that it would be worth looking into a more straight-forward array-based implementation here.

Does that feel like premature optimization to you?

Reply:
You right,
I updated to:
```java
        final List<Filter.ResultItem> resultItems = new ArrayList<>();
        for (Filter filter : linkFilters) {
          final Filter.Result result = filter.applyFilter(message, message.length());
          if (result == null) {
            continue;
          }
          resultItems.addAll(result.getResultItems());
        }
        resultItems.sort(Comparator.comparingInt(Filter.ResultItem::getHighlightStartOffset));
```